taichi/program/program.cpp

```diff
@@ -227,16 +227,15 @@ void Program::check_runtime_error() {
 }
 
 void Program::synchronize() {
-  if (!sync) {
-    if (config.async_mode) {
-      async_engine->synchronize();
-    }
-    if (arch_uses_llvm(config.arch) || config.arch == Arch::metal ||
-        config.arch == Arch::vulkan || config.arch == Arch::opengl) {
-      program_impl_->synchronize();
-    }
+  if (config.async_mode && !sync) {
```

Comment:
Is this part of the plan to remove async..?

Reply:
Yup! I've tried the only blocker for removing this flag is  async tests so it should be gone after the async removal :D